framework/infrastructure/traffic_director.py

```diff
@@ -612,30 +628,54 @@ def create_target_proxy(self):
             target_proxy_type,
             self.url_map.name,
         )
-        self.target_proxy = create_proxy_fn(name, self.url_map)
+        
+        if ipv6:
+            self.target_proxy_v6 = create_proxy_fn(name, self.url_map)
+        else:
+            self.target_proxy = create_proxy_fn(name, self.url_map)
 
-    def delete_target_grpc_proxy(self, force=False):
+    def delete_target_grpc_proxy(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            else:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME + "-v6")
         elif self.target_proxy:
             name = self.target_proxy.name
+        elif self.target_proxy_v6:
+             name = self.target_proxy_v6.name
         else:
             return
+        
         logger.info('Deleting Target GRPC proxy "%s"', name)
         self.compute.delete_target_grpc_proxy(name)
-        self.target_proxy = None
+        if ipv6:
+            self.target_proxy_v6 = None
+        else:
+            self.target_proxy = None
+        
         self.target_proxy_is_http = False
 
-    def delete_target_http_proxy(self, force=False):
+    def delete_target_http_proxy(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            else:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME + "-v6")
         elif self.target_proxy and self.target_proxy_is_http:
             name = self.target_proxy.name
+        elif self.target_proxy_v6 and self.target_proxy_is_http:
+            name = self.target_proxy_v6.name
         else:
             return
```

Comment:
Like in the case above, reduce fragility by wrapping in `if ipv6`.

Reply:
Done

Reply:
I don't see the change.

Reply:
Pushed a new commit. It was overridden due to interchange between cloudtop and local